garnbarn_api/tests/test_view.py

```diff
@@ -217,6 +226,101 @@ def test_delete(self):
         assignments_in_database = Assignment.objects.all()
         self.assertEqual(len(assignments_in_database), 0)
 
+    def test_get_tag(self):
+        """Return detail of the tag."""
+        response = self.client.get("/api/v1/tag/2/")
+        converted_data = convert_to_json(response.content)
+        expected_result = json.dumps({
+            "id": 2,
+            "name": "test_tag2",
+            "author": {
+                'line': None,
+                'name': 'user_name',
+                'uid': 'user_id'
+                },
+            "color": "#4285F4",
+            "reminderTime": None,
+            "subscriber": None
+        })
+        self.assertJSONEqual(expected_result, converted_data)
+        self.assertEqual(200, response.status_code)
+
+    def test_post_tag_without_name(self):
+        """Create tag object without a name."""
+
+        data = {
+            'author': self.user,
+            'color': '#4285F4',
+        }
+        response = self.client.post("/api/v1/tag/", data)
+        converted_data = convert_to_json(response.content)
+        res = json.dumps({
+            "message": {
+                "name": ["This field is required."]
+            }
+        })
+        self.assertJSONEqual(res, converted_data)
+        self.assertEqual(400, response.status_code)
+
+    def test_post_with_invalid_tag_reminder_time(self):
+        """Reminder time contain string"""
+        data = {
+            "name": "test_tag",
+            "reminderTime": [1, 2, "hello"]
+        }
+        response = self.client.post("/api/v1/tag/", data)
+        self.assertEqual(400, response.status_code)
+
+    def test_post_valid_tag(self):
+        """Create tag object with no error."""
+        data = {
+            'name': "test_tag",
+            'author': self.user,
+            'color': '#4285F4',
+            "reminderTime": [3600, 1800],
+            'subscriber': self.user
+        }
+        response = self.client.post("/api/v1/tag/", data)
+        self.assertEqual(200, response.status_code)
```

Comment:
The reminderTime should be sorted before storing in the database

Reply:
Done in commit d34d7a6.